packages/phone-number-privacy/signer/src/pnp/services/request-service.ts

```diff
@@ -0,0 +1,125 @@
+import { ErrorMessage } from '@celo/phone-number-privacy-common'
+import { Knex } from 'knex'
+import { Context } from '../../common/context'
+import { PnpSignRequestRecord } from '../../common/database/models/request'
+import { getPerformedQueryCount, incrementQueryCount } from '../../common/database/wrappers/account'
+import {
+  deleteRequestsOlderThan,
+  getRequestIfExists,
+  insertRequest,
+} from '../../common/database/wrappers/request'
+import { wrapError } from '../../common/error'
+import { Histograms, newMeter } from '../../common/metrics'
+import { traceAsyncFunction } from '../../common/tracing-utils'
+
+export interface PnpRequestService {
+  recordRequest(
+    address: string,
+    blindedQuery: string,
+    signature: string,
+    ctx: Context
+  ): Promise<void>
+  getUsedQuotaForAccount(address: string, ctx: Context): Promise<number>
+  getDuplicateRequest(
+    address: string,
+    blindedQuery: string,
+    ctx: Context
+  ): Promise<PnpSignRequestRecord | undefined>
+  removeOldRequest(daysToKeep: number, ctx: Context): Promise<number>
+}
+
+export class DefaultPnpRequestService implements PnpRequestService {
+  constructor(readonly db: Knex) {}
+
+  public async recordRequest(
+    account: string,
+    blindedQueryPhoneNumber: string,
+    signature: string,
+    ctx: Context
+  ): Promise<void> {
+    return traceAsyncFunction('DefaultPnpRequestService - recordRequest', async () => {
+      return this.db.transaction(async (trx) => {
+        await insertRequest(this.db, account, blindedQueryPhoneNumber, signature, ctx.logger, trx)
+        await incrementQueryCount(this.db, account, ctx.logger, trx)
+      })
+    })
+  }
+
+  public async getUsedQuotaForAccount(account: string, ctx: Context): Promise<number> {
+    const meter = newMeter(
+      Histograms.getRemainingQueryCountInstrumentation,
+      'getQuotaStatus',
+      ctx.url
+    )
+    return traceAsyncFunction('DefaultPnpRequestService - getUsedQuotaForAccount', () =>
+      meter(() =>
+        wrapError(
+          getPerformedQueryCount(this.db, account, ctx.logger),
+          ErrorMessage.FAILURE_TO_GET_PERFORMED_QUERY_COUNT
+        )
+      )
+    )
+  }
+
+  public async getDuplicateRequest(
+    account: string,
+    blindedQueryPhoneNumber: string,
+    ctx: Context
+  ): Promise<PnpSignRequestRecord | undefined> {
+    try {
+      const res = await getRequestIfExists(this.db, account, blindedQueryPhoneNumber, ctx.logger)
+      return res
+    } catch (err) {
+      ctx.logger.error(err, 'Failed to check if request already exists in db')
+      return undefined
+    }
+  }
+
+  public async removeOldRequest(daysToKeep: number, ctx: Context): Promise<number> {
+    if (daysToKeep < 0) {
+      ctx.logger.error('RemoveOldRequest - DaysToKeep should be bigger than or equal to zero')
```

Comment:
If we expect to remove something from the future, this should log and return. Ideally we are going to be the only ones calling this
```suggestion
      ctx.logger.error('RemoveOldRequest - DaysToKeep should be bigger than or equal to zero')
      return 0
```